src/dispatch/group/flows.py

```diff
@@ -118,6 +120,21 @@ def update_group(
             log.exception(e)
             return
 
+        if subject_type == "case":
```

Comment:
Unrelated to the pr, but maybe we could make a `log_subject_event` that takes a `subject_type` param to avoid the double function call where we emit an event.

Reply:
Yeah, I think that's a great idea. There are too many places where we do the same now. We can tackle it in a separate PR.